src/level.ts

```diff
@@ -155,105 +77,109 @@ export function level(useResources = true) {
-      setMode(MODE_MACRO);
-      Macro.externalIf(useGg && haveEffect($effect`Giant Growth`) === 0, 'skill Giant Growth')
+
+      Macro.externalIf(useGg && !have($effect`Giant Growth`), Macro.skill($skill`Giant Growth`))
         .kill()
         .save();
+      visitUrl("main.php?fightgodlobster=1");
       runCombat();
-      visitUrl('choice.php');
+      visitUrl("choice.php");
       if (handlingChoice()) runChoice(3);
-      setMode(MODE_NULL);
     }
   }
 
+  leaflet();
+
   if (
-    getPropertyInt('_sausageFights') === 0 &&
-    haveFamiliar($familiar`Pocket Professor`) &&
-    availableAmount($item`Kramco Sausage-o-Matic&trade;`) > 0
+    get("_sausageFights") === 0 &&
+    have($familiar`Pocket Professor`) &&
+    have($item`Kramco Sausage-o-Matic™`)
   ) {
     useFamiliar($familiar`Pocket Professor`);
     maximizeCached(
-      'mainstat, 4 exp, 30 mainstat experience percent, 30 familiar weight, equip makeshift garbage shirt, equip Pocket Professor memory chip, equip Kramco'
+      ["Mainstat", "4 Experience", "30 Mainstat Experience Percent", "30 Familiar Weight"],
+      {
+        forceEquip: $items`makeshift garbage shirt, Pocket Professor memory chip`,
+      }
     );
-    levelMood();
+    moodLevel().execute();
+    restoreMp(100);
     restoreHp(myMaxhp());
-    tryEnsureEffect($effect`Chorale of Companionship`);
-    tryEnsureEffect($effect`Billiards Belligerence`);
-    tryEnsureEffect($effect`Do I Know You From Somewhere?`);
-    tryEnsureEffect($effect`Oiled, Slick`) || tryEnsureEffect($effect`Oiled Up`);
-    adventureCopy($location`"The Outskirts of Cobb's Knob"`, $monster`sausage goblin`);
+
+    tryEnsureEffect($effect`Oiled, Slick`) || tryEnsureEffect($effect`Oiled-Up`);
+    adventureMacro(
+      $location`The Outskirts of Cobb's Knob`,
+      Macro.if_("!monstername sausage goblin", Macro.abort())
+        .skill($skill`lecture on relativity`)
+        .kill()
+    );
   }
 
-  while (getPropertyInt('_neverendingPartyFreeTurns') < 10) {
-    if (!getPropertyBoolean('leafletCompleted') && myLevel() >= 9) {
-      visitUrl('council.php');
-      cliExecute('leaflet');
-    }
+  while (get("_neverendingPartyFreeTurns") < 10) {
+    leaflet();
+    propertyManager.setChoices({ 1324: 5 });
     useFamiliar($familiar`Hovering Sombrero`);
-    maximizeCached('mainstat, 4exp, equip makeshift garbage shirt');
-    setChoice(1324, 5);
-    levelMood();
+    maximizeCached(["Mainstat", "4 Experience"], { forceEquip: $items`makeshift garbage shirt` });
+    moodLevel().execute();
+    restoreMp(100);
+    restoreHp(myMaxhp());
+
     adventureMacro($location`The Neverending Party`, Macro.kill());
   }
 
-  visitUrl('council.php');
+  visitUrl("council.php");
 
-  print('');
-  print('Done leveling.', 'blue');
-  print(`Reached mainstat ${myBasestat(myPrimestat())}`);
+  print("");
+  print("Done leveling.", "blue");
+  print(`Reached mainstat ${myBasestat(myPrimestat())}.`);
+  print(`Reached level ${myLevel()}.`);
```

Comment:
Do we want to throw if this number is small?

Reply:
we throw if it's <13